### ISSUE TITLE
Upgrade to composer-plugin-api 2.0.0

### ISSUE THREAD

**Issue by chaos0815** [laminas/laminas-dependency-plugin#21]
In order to take advantage of composer 2.0 the version in composer.json needs to be bumped up.

**Comment by FabianKoestring** [comment]
@chaos0815 There is already a working Beta-Release (https://github.com/laminas/laminas-dependency-plugin/releases/tag/2.0.0beta1). But it would be nice to tag a final version.

**Comment by boesing** [comment]
I'll create a `final` version when I finished cleaning up some stuff AND when there is more Feedback. Thats why its called beta as I want to have tested more scenarios than just the project I've worked on while implementing the new process.

**Comment by driehle** [comment]
@boesing Given that Composer v2 is now released and installed by default, I think that releasing a compatible package here is quite important. With Composer v2 my pielines break, because it finds not install candidate. Adding `"minimum-stability": "beta"` solves this. What type of feedback do you expect? For my Laminas projects Composer v2 runs quite smoothly, though I probably have migrated everything to Laminas by now, so I am not sure if there is a need for this plugin to intercept anymore.

**Comment by jonathanmaron** [comment]
@boesing After upgrading to Composer 2.0, I updated all my applications to 2.0.0beta1 yesterday afternoon, and following substantial testing since then, cannot find any issues whatsoever. I can only re-iterate what @driehle already stated: "I think that releasing a compatible package here is quite important" :-) Thank you for your hard work.

**Comment by codyloco** [comment]
@jonathanmaron did you upgrade to beta just by adding minimum-stability: "beta"?  I'm struggling to upgrade to Magento 2.4.1 because of this dependency not working with Composer 2.

**Comment by weierophinney** [comment]
Just a note: you don't need to set the minimum stability. Each dependency you specify can include the stability allowed by adding it following an `@` symbol:

```json
    "laminas/laminas-dependency-plugin": "~2.0.0beta1@beta",
```

**Comment by jonathanmaron** [comment]
@codyloco The line in my `composer.json` is like this:

`"laminas/laminas-dependency-plugin": "2.0.0beta1",`

**Comment by codyloco** [comment]
Thanks guys, if something burns down I'll let ya know :)

**Comment by weierophinney** [comment]
I've released version 2.0.0 at this time.
